src/interfaces.py

```diff
@@ -683,11 +683,16 @@ def is_call(inst: Instruction) -> bool:
 
 class Generator(ABC):
     instruction_set: InstructionSetAbstract
+    _state: int = 0
 
-    def __init__(self, instruction_set: InstructionSetAbstract):
+    def __init__(self, instruction_set: InstructionSetAbstract, seed: int):
         self.instruction_set = instruction_set
+        self.set_seed(seed)
         super().__init__()
 
+    def set_seed(self, seed: int):
+        self._state = seed
```

Comment:
Just an idea: would there ever be the need for a `self.get_seed()` so other modules/classes don't have to access the internal `_state` field? It functionally makes no difference, but it might be a cleaner way to access the seed. But, this may not be needed.

Reply:
Makes sense